scripts/squad-config

```diff
@@ -0,0 +1,604 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import netrc
+import os
+import requests
+import sys
+import yaml
+
+from copy import deepcopy
+from jsonobject import *
+from jsonobject.base import get_dynamic_properties
+from pprint import pprint
+from urllib.parse import urlsplit, urlunsplit
+
+
+FORMAT = "%(funcName)20s() ] %(message)s"
+logging.basicConfig(level=logging.INFO, format=FORMAT)
+logger = logging.getLogger(__name__)
+
+
+# ExtLoaderMeta and ExtLoader copied from
+# https://gist.github.com/joshbode/569627ced3076931b02f
+class ExtLoaderMeta(type):
+
+    def __new__(metacls, __name__, __bases__, __dict__):
+        """Add include constructer to class."""
+
+        # register the include constructor on the class
+        cls = super().__new__(metacls, __name__, __bases__, __dict__)
+        cls.add_constructor('!include', cls.construct_include)
+
+        return cls
+
+
+class ExtLoader(yaml.Loader, metaclass=ExtLoaderMeta):
+    """YAML Loader with `!include` constructor."""
+
+    def __init__(self, stream):
+        """Initialise Loader."""
+
+        try:
+            self._root = os.path.split(stream.name)[0]
+        except AttributeError:
+            self._root = os.path.curdir
+
+        super().__init__(stream)
+
+    def construct_include(self, node):
+        """Include file referenced at node."""
+
+        filename = os.path.abspath(os.path.join(
+            self._root, self.construct_scalar(node)
+        ))
+        extension = os.path.splitext(filename)[1].lstrip('.')
+
+        with open(filename, 'r') as f:
+            if extension in ('yaml', 'yml'):
+                return yaml.load(f, ExtLoader)
+            else:
+                return ''.join(f.readlines())
+
+
+def yaml_validator(data):
+    yaml.load(data, Loader=ExtLoader)
+
+
+class SquadConnectionException(Exception):
+    pass
+
+
+class SquadConnection(object):
+    def __init__(self, args):
+        self.url = args.url
+        urlparts = urlsplit(self.url)
+        self.base_url = urlparts.netloc
+        self.url_scheme = urlparts.scheme
+
+        connection_token = "Token %s" % self.__get_connection_token__(args.url)
+        self.headers = {
+            "Authorization": connection_token
+        }
+
+    def get_prepared_request(self, endpoint, method):
+        URL = urlunsplit((self.url_scheme, self.base_url, "api/%s" % endpoint, None, None))
+        req = requests.Request(method, URL, headers=self.headers)
+        return req.prepare()
+
+    def __get_connection_token__(self, url):
+        netrcauth = netrc.netrc()
+        try:
+            self.username, _, self.token = netrcauth.authenticators(self.base_url)
+            logger.info("Using username: %s" % self.username)
+            return self.token
+        except TypeError:
+            logger.error("No credentials found for %s" % self.base_url)
```

Comment:
it should exit here; if a first attempt to get credentials don't work, there is no point in continuing

Reply:
ACK